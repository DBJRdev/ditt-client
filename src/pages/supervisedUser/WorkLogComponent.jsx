import ImmutablePropTypes from 'react-immutable-proptypes';
import PropTypes from 'prop-types';
import React from 'react';
import { withTranslation } from 'react-i18next';
import jwt from 'jsonwebtoken';
import WorkLogCalendar from '../../components/WorkLogCalendar';
import Layout from '../../components/Layout';
import {
  STATUS_APPROVED,
  STATUS_OPENED,
  STATUS_WAITING_FOR_APPROVAL,
} from '../../resources/workMonth';
import {
  createDate,
  getAllDays,
  localizedMoment,
} from '../../services/dateTimeService';
import { getWorkMonthByMonth } from '../../services/workLogService';

class WorkLogComponent extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      selectedDate: localizedMoment(),
    };

    if (this.props.match.params.year && this.props.match.params.month) {
      this.state = {
        selectedDate: createDate(
          this.props.match.params.year,
          this.props.match.params.month - 1,
          1,
        ),
      };
    }

    this.changeSelectedDate = this.changeSelectedDate.bind(this);
    this.addMultipleMaternityProtectionWorkLogs = this.addMultipleMaternityProtectionWorkLogs
      .bind(this);
    this.deleteMultipleMaternityProtectionWorkLogs = this.deleteMultipleMaternityProtectionWorkLogs
      .bind(this);
  }

  componentDidMount() {
    this.props.fetchConfig();
    this.props.fetchWorkHoursList(this.props.match.params.id);
    this.props.fetchWorkMonthList(this.props.match.params.id).then(() => {
      this.fetchWorkMonth(this.state.selectedDate);
    });
  }

  fetchWorkMonth(selectedDate) {
    const workMonth = getWorkMonthByMonth(selectedDate, this.props.workMonthList.toJS());

    if (workMonth) {
      return this.props.fetchWorkMonth(workMonth.id);
    }

    return null;
  }

  changeSelectedDate(selectedDate) {
    this.fetchWorkMonth(selectedDate).then(() => this.setState({ selectedDate }));
  }

  addMultipleMaternityProtectionWorkLogs(data) {
    const { addMultipleMaternityProtectionWorkLogs } = this.props;
    const workingDays = getAllDays(data.date, data.dateTo);
    const workLogs = workingDays.map((workingDay) => ({ date: workingDay }));

    return addMultipleMaternityProtectionWorkLogs({
      user: data.user,
      workLogs,
    }).then((response) => {
      if (!response.error) {
        this.fetchWorkMonth(this.state.selectedDate);
      }

      return response;
    });
  }

  deleteMultipleMaternityProtectionWorkLogs(id) {
    return this.props.deleteMaternityProtectionWorkLog(id).then((response) => {
      if (!response.error) {
        this.fetchWorkMonth(this.state.selectedDate);
      }

      return response;
    });
  }

  render() {
    let title = this.props.t('workLog:title.workLogs');

    if (this.props.workMonth) {
      const user = this.props.workMonth.get('user');
      const name = `${user.get('firstName')} ${user.get('lastName')}`;
      title = this.props.t('workLog:title.supervisedUserWorkLogs', { name });
    }

    let uid = null;

    if (this.props.token) {
      const decodedToken = jwt.decode(this.props.token);

      if (decodedToken) {
        // eslint-disable-next-line prefer-destructuring
        uid = decodedToken.uid;
      }
    }

    return (
      <Layout title={title} loading={this.props.isFetching}>
        {this.props.config && (
          <WorkLogCalendar
            addBusinessTripWorkLog={() => {}}
            addHomeOfficeWorkLog={() => {}}
            addOvertimeWorkLog={() => {}}
            addSickDayWorkLog={() => {}}
            addTimeOffWorkLog={() => {}}
            addMultipleMaternityProtectionWorkLogs={this.addMultipleMaternityProtectionWorkLogs}
            addMultipleVacationWorkLog={() => {}}
            addWorkLog={() => {}}
            businessTripWorkLog={this.props.businessTripWorkLog}
            changeSelectedDate={this.changeSelectedDate}
            config={this.props.config}
            deleteBusinessTripWorkLog={() => {}}
            deleteHomeOfficeWorkLog={() => {}}
            deleteMaternityProtectionWorkLog={this.deleteMultipleMaternityProtectionWorkLogs}
            deleteOvertimeWorkLog={() => {}}
            deleteSickDayWorkLog={() => {}}
            deleteTimeOffWorkLog={() => {}}
            deleteVacationWorkLog={() => {}}
            deleteWorkLog={() => {}}
            fetchBusinessTripWorkLog={this.props.fetchBusinessTripWorkLog}
            fetchHomeOfficeWorkLog={this.props.fetchHomeOfficeWorkLog}
            fetchMaternityProtectionWorkLog={this.props.fetchMaternityProtectionWorkLog}
            fetchOvertimeWorkLog={this.props.fetchOvertimeWorkLog}
            fetchSickDayWorkLog={this.props.fetchSickDayWorkLog}
            fetchTimeOffWorkLog={this.props.fetchTimeOffWorkLog}
            fetchVacationWorkLog={this.props.fetchVacationWorkLog}
            fetchWorkLog={this.props.fetchWorkLog}
            homeOfficeWorkLog={this.props.homeOfficeWorkLog}
            isPosting={this.props.isPosting}
            markApproved={this.props.markApproved}
            markWaitingForApproval={() => {}}
            maternityProtectionWorkLog={this.props.maternityProtectionWorkLog}
            overtimeWorkLog={this.props.overtimeWorkLog}
            selectedDate={this.state.selectedDate}
            supervisorView
            sickDayWorkLog={this.props.sickDayWorkLog}
            timeOffWorkLog={this.props.timeOffWorkLog}
            uid={uid}
            vacationWorkLog={this.props.vacationWorkLog}
            workHoursList={this.props.workHoursList}
            workLog={this.props.workLog}
            workMonth={this.props.workMonth}
            workMonthList={this.props.workMonthList}
          />
        )}
      </Layout>
    );
  }
}

WorkLogComponent.defaultProps = {
  businessTripWorkLog: null,
  config: {},
  homeOfficeWorkLog: null,
  maternityProtectionWorkLog: null,
  overtimeWorkLog: null,
  sickDayWorkLog: null,
  timeOffWorkLog: null,
  vacationWorkLog: null,
  workLog: null,
  workMonth: null,
};

WorkLogComponent.propTypes = {
  addMultipleMaternityProtectionWorkLogs: PropTypes.func.isRequired,
  businessTripWorkLog: ImmutablePropTypes.mapContains({
    date: PropTypes.object.isRequired,
    destination: PropTypes.string.isRequired,
    expectedArrival: PropTypes.string.isRequired,
    expectedDeparture: PropTypes.string.isRequired,
    purpose: PropTypes.string.isRequired,
    rejectionMessage: PropTypes.string,
    status: PropTypes.string.isRequired,
    transport: PropTypes.string.isRequired,
  }),
  config: ImmutablePropTypes.mapContains({}),
  deleteMaternityProtectionWorkLog: PropTypes.func.isRequired,
  fetchBusinessTripWorkLog: PropTypes.func.isRequired,
  fetchConfig: PropTypes.func.isRequired,
  fetchHomeOfficeWorkLog: PropTypes.func.isRequired,
  fetchMaternityProtectionWorkLog: PropTypes.func.isRequired,
  fetchOvertimeWorkLog: PropTypes.func.isRequired,
  fetchSickDayWorkLog: PropTypes.func.isRequired,
  fetchTimeOffWorkLog: PropTypes.func.isRequired,
  fetchVacationWorkLog: PropTypes.func.isRequired,
  fetchWorkHoursList: PropTypes.func.isRequired,
  fetchWorkLog: PropTypes.func.isRequired,
  fetchWorkMonth: PropTypes.func.isRequired,
  fetchWorkMonthList: PropTypes.func.isRequired,
  homeOfficeWorkLog: ImmutablePropTypes.mapContains({
    comment: PropTypes.string,
    date: PropTypes.object.isRequired,
    rejectionMessage: PropTypes.string,
    status: PropTypes.string.isRequired,
  }),
  isFetching: PropTypes.bool.isRequired,
  isPosting: PropTypes.bool.isRequired,
  markApproved: PropTypes.func.isRequired,
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string.isRequired,
      month: PropTypes.string,
      year: PropTypes.string,
    }).isRequired,
  }).isRequired,
  maternityProtectionWorkLog: ImmutablePropTypes.mapContains({
    date: PropTypes.object.isRequired,
  }),
  overtimeWorkLog: ImmutablePropTypes.mapContains({
    date: PropTypes.object.isRequired,
    reason: PropTypes.string,
    rejectionMessage: PropTypes.string,
    status: PropTypes.string.isRequired,
  }),
  sickDayWorkLog: ImmutablePropTypes.mapContains({
    childDateOfBirth: PropTypes.object,
    childName: PropTypes.string,
    date: PropTypes.object.isRequired,
    variant: PropTypes.string.isRequired,
  }),
  t: PropTypes.func.isRequired,
  timeOffWorkLog: ImmutablePropTypes.mapContains({
    comment: PropTypes.string,
    date: PropTypes.object.isRequired,
    rejectionMessage: PropTypes.string,
    status: PropTypes.string.isRequired,
  }),
  token: PropTypes.string.isRequired,
  vacationWorkLog: ImmutablePropTypes.mapContains({
    date: PropTypes.object.isRequired,
    rejectionMessage: PropTypes.string,
    status: PropTypes.string.isRequired,
  }),
  workHoursList: ImmutablePropTypes.listOf(ImmutablePropTypes.mapContains({
    month: PropTypes.number.isRequired,
    requiredHours: PropTypes.number.isRequired,
    year: PropTypes.number.isRequired,
  })).isRequired,
  workLog: ImmutablePropTypes.mapContains({
    endTime: PropTypes.object.isRequired,
    startTime: PropTypes.object.isRequired,
  }),
  workMonth: ImmutablePropTypes.mapContains({
    id: PropTypes.number.isRequired,
    month: PropTypes.shape.isRequired,
    status: PropTypes.oneOf([
      STATUS_APPROVED,
      STATUS_OPENED,
      STATUS_WAITING_FOR_APPROVAL,
    ]).isRequired,
    user: ImmutablePropTypes.mapContains({
      firstName: PropTypes.string.isRequired,
      lastName: PropTypes.string.isRequired,
    }),
    workLogs: ImmutablePropTypes.listOf(ImmutablePropTypes.mapContains({
      endTime: PropTypes.shape.isRequired,
      id: PropTypes.number.isRequired,
      startTime: PropTypes.shape.isRequired,
    })).isRequired,
    year: PropTypes.number,
  }),
  workMonthList: ImmutablePropTypes.listOf(ImmutablePropTypes.mapContains({
    id: PropTypes.number.isRequired,
    month: PropTypes.shape.isRequired,
    year: PropTypes.number.isRequired,
  })).isRequired,
};

export default withTranslation()(WorkLogComponent);
