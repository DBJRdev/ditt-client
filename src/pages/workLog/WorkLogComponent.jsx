import ImmutablePropTypes from 'react-immutable-proptypes';
import PropTypes from 'prop-types';
import React from 'react';
import { withNamespaces } from 'react-i18next';
import WorkLogCalendar from '../../components/WorkLogCalendar';
import Layout from '../../components/Layout';
import {
  STATUS_APPROVED,
  STATUS_OPENED,
  STATUS_WAITING_FOR_APPROVAL,
} from '../../resources/workMonth';
import { localizedMoment } from '../../services/dateTimeService';
import { getWorkMonthByMonth } from '../../services/workLogService';

class WorkLogComponent extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      selectedDate: localizedMoment(),
    };

    this.addBusinessTripWorkLog = this.addBusinessTripWorkLog.bind(this);
    this.addHomeOfficeWorkLog = this.addHomeOfficeWorkLog.bind(this);
    this.addOvertimeWorkLog = this.addOvertimeWorkLog.bind(this);
    this.addSickDayWorkLog = this.addSickDayWorkLog.bind(this);
    this.addTimeOffWorkLog = this.addTimeOffWorkLog.bind(this);
    this.addVacationWorkLog = this.addVacationWorkLog.bind(this);
    this.addWorkLog = this.addWorkLog.bind(this);
    this.deleteBusinessTripWorkLog = this.deleteBusinessTripWorkLog.bind(this);
    this.deleteHomeOfficeWorkLog = this.deleteHomeOfficeWorkLog.bind(this);
    this.deleteOvertimeWorkLog = this.deleteOvertimeWorkLog.bind(this);
    this.deleteSickDayWorkLog = this.deleteSickDayWorkLog.bind(this);
    this.deleteTimeOffWorkLog = this.deleteTimeOffWorkLog.bind(this);
    this.deleteVacationWorkLog = this.deleteVacationWorkLog.bind(this);
    this.deleteWorkLog = this.deleteWorkLog.bind(this);
    this.changeSelectedDate = this.changeSelectedDate.bind(this);
  }

  componentDidMount() {
    this.props.fetchConfig();
    this.props.fetchWorkHoursList(this.props.uid);
    this.props.fetchWorkMonthList(this.props.uid).then(() => {
      this.fetchWorkMonth(this.state.selectedDate);
    });
  }

  addBusinessTripWorkLog(data) {
    return this.props.addBusinessTripWorkLog(data).then((response) => {
      if (!response.error) {
        this.fetchWorkMonth(this.state.selectedDate);
      }

      return response;
    });
  }

  addHomeOfficeWorkLog(data) {
    return this.props.addHomeOfficeWorkLog(data).then((response) => {
      if (!response.error) {
        this.fetchWorkMonth(this.state.selectedDate);
      }

      return response;
    });
  }

  addOvertimeWorkLog(data) {
    return this.props.addOvertimeWorkLog(data).then((response) => {
      if (!response.error) {
        this.fetchWorkMonth(this.state.selectedDate);
      }

      return response;
    });
  }

  addSickDayWorkLog(data) {
    return this.props.addSickDayWorkLog(data).then((response) => {
      if (!response.error) {
        this.fetchWorkMonth(this.state.selectedDate);
      }

      return response;
    });
  }

  addTimeOffWorkLog(data) {
    return this.props.addTimeOffWorkLog(data).then((response) => {
      if (!response.error) {
        this.fetchWorkMonth(this.state.selectedDate);
      }
      return response;
    });
  }

  addVacationWorkLog(data) {
    return this.props.addVacationWorkLog(data).then((response) => {
      if (!response.error) {
        this.fetchWorkMonth(this.state.selectedDate);
      }
      return response;
    });
  }

  addWorkLog(data) {
    return this.props.addWorkLog(data).then((response) => {
      if (!response.error) {
        this.fetchWorkMonth(this.state.selectedDate);
      }

      return response;
    });
  }

  deleteBusinessTripWorkLog(id) {
    return this.props.deleteBusinessTripWorkLog(id).then((response) => {
      if (!response.error) {
        this.fetchWorkMonth(this.state.selectedDate);
      }

      return response;
    });
  }

  deleteHomeOfficeWorkLog(id) {
    return this.props.deleteHomeOfficeWorkLog(id).then((response) => {
      if (!response.error) {
        this.fetchWorkMonth(this.state.selectedDate);
      }

      return response;
    });
  }

  deleteOvertimeWorkLog(id) {
    return this.props.deleteOvertimeWorkLog(id).then((response) => {
      if (!response.error) {
        this.fetchWorkMonth(this.state.selectedDate);
      }

      return response;
    });
  }

  deleteSickDayWorkLog(id) {
    return this.props.deleteSickDayWorkLog(id).then((response) => {
      if (!response.error) {
        this.fetchWorkMonth(this.state.selectedDate);
      }

      return response;
    });
  }

  deleteTimeOffWorkLog(id) {
    return this.props.deleteTimeOffWorkLog(id).then((response) => {
      if (!response.error) {
        this.fetchWorkMonth(this.state.selectedDate);
      }

      return response;
    });
  }

  deleteVacationWorkLog(id) {
    return this.props.deleteVacationWorkLog(id).then((response) => {
      if (!response.error) {
        this.fetchWorkMonth(this.state.selectedDate);
      }

      return response;
    });
  }

  deleteWorkLog(id) {
    return this.props.deleteWorkLog(id).then((response) => {
      if (!response.error) {
        this.fetchWorkMonth(this.state.selectedDate);
      }

      return response;
    });
  }

  fetchWorkMonth(selectedDate) {
    const workMonth = getWorkMonthByMonth(selectedDate, this.props.workMonthList.toJS());

    if (workMonth) {
      return this.props.fetchWorkMonth(workMonth.id);
    }

    return null;
  }

  changeSelectedDate(selectedDate) {
    this.fetchWorkMonth(selectedDate).then(() => this.setState({ selectedDate }));
  }

  render() {
    return (
      <Layout title={this.props.t('workLog:title.workLogs')} loading={this.props.isFetching}>
        {this.props.config && (
          <WorkLogCalendar
            addBusinessTripWorkLog={this.addBusinessTripWorkLog}
            addHomeOfficeWorkLog={this.addHomeOfficeWorkLog}
            addOvertimeWorkLog={this.addOvertimeWorkLog}
            addSickDayWorkLog={this.addSickDayWorkLog}
            addTimeOffWorkLog={this.addTimeOffWorkLog}
            addVacationWorkLog={this.addVacationWorkLog}
            addWorkLog={this.addWorkLog}
            businessTripWorkLog={this.props.businessTripWorkLog}
            changeSelectedDate={this.changeSelectedDate}
            config={this.props.config}
            deleteBusinessTripWorkLog={this.deleteBusinessTripWorkLog}
            deleteHomeOfficeWorkLog={this.deleteHomeOfficeWorkLog}
            deleteOvertimeWorkLog={this.deleteOvertimeWorkLog}
            deleteSickDayWorkLog={this.deleteSickDayWorkLog}
            deleteTimeOffWorkLog={this.deleteTimeOffWorkLog}
            deleteVacationWorkLog={this.deleteVacationWorkLog}
            deleteWorkLog={this.deleteWorkLog}
            fetchBusinessTripWorkLog={this.props.fetchBusinessTripWorkLog}
            fetchHomeOfficeWorkLog={this.props.fetchHomeOfficeWorkLog}
            fetchOvertimeWorkLog={this.props.fetchOvertimeWorkLog}
            fetchSickDayWorkLog={this.props.fetchSickDayWorkLog}
            fetchTimeOffWorkLog={this.props.fetchTimeOffWorkLog}
            fetchVacationWorkLog={this.props.fetchVacationWorkLog}
            fetchWorkLog={this.props.fetchWorkLog}
            homeOfficeWorkLog={this.props.homeOfficeWorkLog}
            isPosting={this.props.isPosting}
            markApproved={() => {}}
            markWaitingForApproval={this.props.markWaitingForApproval}
            overtimeWorkLog={this.props.overtimeWorkLog}
            selectedDate={this.state.selectedDate}
            sickDayWorkLog={this.props.sickDayWorkLog}
            timeOffWorkLog={this.props.timeOffWorkLog}
            vacationWorkLog={this.props.vacationWorkLog}
            workHoursList={this.props.workHoursList}
            workLog={this.props.workLog}
            workMonth={this.props.workMonth}
            workMonthList={this.props.workMonthList}
          />
        )}
      </Layout>
    );
  }
}

WorkLogComponent.defaultProps = {
  businessTripWorkLog: null,
  config: {},
  homeOfficeWorkLog: null,
  overtimeWorkLog: null,
  sickDayWorkLog: null,
  timeOffWorkLog: null,
  uid: null,
  vacationWorkLog: null,
  workLog: null,
  workMonth: null,
};

WorkLogComponent.propTypes = {
  addBusinessTripWorkLog: PropTypes.func.isRequired,
  addHomeOfficeWorkLog: PropTypes.func.isRequired,
  addOvertimeWorkLog: PropTypes.func.isRequired,
  addSickDayWorkLog: PropTypes.func.isRequired,
  addTimeOffWorkLog: PropTypes.func.isRequired,
  addVacationWorkLog: PropTypes.func.isRequired,
  addWorkLog: PropTypes.func.isRequired,
  businessTripWorkLog: ImmutablePropTypes.mapContains({
    date: PropTypes.object.isRequired,
    destination: PropTypes.string.isRequired,
    expectedArrival: PropTypes.string.isRequired,
    expectedDeparture: PropTypes.string.isRequired,
    purpose: PropTypes.string.isRequired,
    rejectionMessage: PropTypes.string,
    status: PropTypes.string.isRequired,
    transport: PropTypes.string.isRequired,
  }),
  config: ImmutablePropTypes.mapContains({}),
  deleteBusinessTripWorkLog: PropTypes.func.isRequired,
  deleteHomeOfficeWorkLog: PropTypes.func.isRequired,
  deleteOvertimeWorkLog: PropTypes.func.isRequired,
  deleteSickDayWorkLog: PropTypes.func.isRequired,
  deleteTimeOffWorkLog: PropTypes.func.isRequired,
  deleteVacationWorkLog: PropTypes.func.isRequired,
  deleteWorkLog: PropTypes.func.isRequired,
  fetchBusinessTripWorkLog: PropTypes.func.isRequired,
  fetchConfig: PropTypes.func.isRequired,
  fetchHomeOfficeWorkLog: PropTypes.func.isRequired,
  fetchOvertimeWorkLog: PropTypes.func.isRequired,
  fetchSickDayWorkLog: PropTypes.func.isRequired,
  fetchTimeOffWorkLog: PropTypes.func.isRequired,
  fetchVacationWorkLog: PropTypes.func.isRequired,
  fetchWorkHoursList: PropTypes.func.isRequired,
  fetchWorkLog: PropTypes.func.isRequired,
  fetchWorkMonth: PropTypes.func.isRequired,
  fetchWorkMonthList: PropTypes.func.isRequired,
  homeOfficeWorkLog: ImmutablePropTypes.mapContains({
    date: PropTypes.object.isRequired,
    rejectionMessage: PropTypes.string,
    status: PropTypes.string.isRequired,
  }),
  isFetching: PropTypes.bool.isRequired,
  isPosting: PropTypes.bool.isRequired,
  markWaitingForApproval: PropTypes.func.isRequired,
  overtimeWorkLog: ImmutablePropTypes.mapContains({
    date: PropTypes.object.isRequired,
    reason: PropTypes.string,
    rejectionMessage: PropTypes.string,
    status: PropTypes.string.isRequired,
  }),
  sickDayWorkLog: ImmutablePropTypes.mapContains({
    childDateOfBirth: PropTypes.object,
    childName: PropTypes.string,
    date: PropTypes.object.isRequired,
    variant: PropTypes.string.isRequired,
  }),
  t: PropTypes.func.isRequired,
  timeOffWorkLog: ImmutablePropTypes.mapContains({
    date: PropTypes.object.isRequired,
    rejectionMessage: PropTypes.string,
    status: PropTypes.string.isRequired,
  }),
  uid: PropTypes.number,
  vacationWorkLog: ImmutablePropTypes.mapContains({
    date: PropTypes.object.isRequired,
    rejectionMessage: PropTypes.string,
    status: PropTypes.string.isRequired,
  }),
  workHoursList: ImmutablePropTypes.listOf(ImmutablePropTypes.mapContains({
    month: PropTypes.number.isRequired,
    requiredHours: PropTypes.number.isRequired,
    year: PropTypes.number.isRequired,
  })).isRequired,
  workLog: ImmutablePropTypes.mapContains({
    endTime: PropTypes.object.isRequired,
    startTime: PropTypes.object.isRequired,
  }),
  workMonth: ImmutablePropTypes.mapContains({
    id: PropTypes.number.isRequired,
    month: PropTypes.shape.isRequired,
    status: PropTypes.oneOf([
      STATUS_APPROVED,
      STATUS_OPENED,
      STATUS_WAITING_FOR_APPROVAL,
    ]).isRequired,
    workLogs: ImmutablePropTypes.listOf(ImmutablePropTypes.mapContains({
      endTime: PropTypes.shape.isRequired,
      id: PropTypes.number.isRequired,
      startTime: PropTypes.shape.isRequired,
    })).isRequired,
    year: PropTypes.number,
  }),
  workMonthList: ImmutablePropTypes.listOf(ImmutablePropTypes.mapContains({
    id: PropTypes.number.isRequired,
    month: PropTypes.shape.isRequired,
    year: PropTypes.number.isRequired,
  })).isRequired,
};

export default withNamespaces()(WorkLogComponent);
