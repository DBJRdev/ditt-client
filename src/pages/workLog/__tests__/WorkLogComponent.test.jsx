import Immutable from 'immutable';
import React from 'react';
import sinon from 'sinon';
import { shallow } from 'enzyme';
import WorkLogComponent from '../WorkLogComponent';

let fakeTimestamp = null;

beforeEach(() => {
  fakeTimestamp = sinon.useFakeTimers(1514764800000);
});

afterEach(() => {
  fakeTimestamp.restore();
});

describe('rendering', () => {
  it('renders correctly', () => {
    const props = {
      addWorkLog: () => {},
      deleteWorkLog: () => {},
      fetchWorkHoursList: () => {},
      fetchWorkMonth: () => {},
      fetchWorkMonthList: () => new Promise(() => {}),
      history: {
        push: () => {},
      },
      isFetching: false,
      isPosting: false,
      markWaitingForApproval: () => {},
      uid: 1,
      workHoursList: Immutable.List(),
      workMonth: null,
      workMonthList: Immutable.List(),
    };
    const tree = shallow(<WorkLogComponent {...props} />);

    expect(tree).toMatchSnapshot();
  });
});
