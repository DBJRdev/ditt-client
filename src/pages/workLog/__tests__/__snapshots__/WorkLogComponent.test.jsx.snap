// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders correctly 1`] = `
<Connect(LayoutComponent)
  loading={false}
  title="workLog:title.workLogs"
>
  <WorkLogCalendar
    addBusinessTripWorkLog={[Function]}
    addHomeOfficeWorkLog={[Function]}
    addMultipleVacationWorkLog={[Function]}
    addOvertimeWorkLog={[Function]}
    addSickDayWorkLog={[Function]}
    addTimeOffWorkLog={[Function]}
    addWorkLog={[Function]}
    businessTripWorkLog={null}
    changeSelectedDate={[Function]}
    config={
      Immutable.Map {
        "supportedHolidays": Immutable.List [
          "2018-01-01T00:00:00.000Z",
          "2018-12-24T00:00:00.000Z",
          "2018-12-25T00:00:00.000Z",
          "2018-12-26T00:00:00.000Z",
          "2018-12-31T00:00:00.000Z",
        ],
        "supportedYears": Immutable.List [
          2018,
          2019,
          2020,
          2021,
        ],
        "workedHoursLimits": Immutable.Map {
          "lowerLimit": Immutable.Map {
            "changeBy": -1800,
            "limit": 21600,
          },
          "upperLimit": Immutable.Map {
            "changeBy": -2700,
            "limit": 32400,
          },
        },
      }
    }
    deleteBusinessTripWorkLog={[Function]}
    deleteHomeOfficeWorkLog={[Function]}
    deleteOvertimeWorkLog={[Function]}
    deleteSickDayWorkLog={[Function]}
    deleteTimeOffWorkLog={[Function]}
    deleteVacationWorkLog={[Function]}
    deleteWorkLog={[Function]}
    fetchBusinessTripWorkLog={[Function]}
    fetchHomeOfficeWorkLog={[Function]}
    fetchOvertimeWorkLog={[Function]}
    fetchSickDayWorkLog={[Function]}
    fetchTimeOffWorkLog={[Function]}
    fetchVacationWorkLog={[Function]}
    fetchWorkLog={[Function]}
    homeOfficeWorkLog={null}
    isPosting={false}
    markApproved={[Function]}
    markWaitingForApproval={[Function]}
    overtimeWorkLog={null}
    selectedDate={"2018-01-01T00:00:00.000Z"}
    sickDayWorkLog={null}
    supervisorView={false}
    t={[Function]}
    timeOffWorkLog={null}
    vacationWorkLog={null}
    workHoursList={Immutable.List []}
    workLog={null}
    workMonth={null}
    workMonthList={Immutable.List []}
  />
</Connect(LayoutComponent)>
`;
